### DISCUSSION TITLE
Сборка с использованием MSYS2/mingw64

### DISCUSSION THREAD

**Discussion by AnSMru** [QuasarApp/CQtDeployer#769] [Q&A] [answered]
Я использую библиотеки от MSYS2 (QT5.15.10, QT6.5.1)
Проблема при использовании qt6.5.1, моя строка для сборки
"%CQtDeployer%" -bin name_prg.exe clear deploySystem allQmlDependes
1. Сборка проходит отлично, только в bat-файле каждый раз приходится править 
SET PATH=%BASE_DIR%\;%BASE_DIR%\;%PATH%;\systemLibs
на SET PATH=%BASE_DIR%\;%BASE_DIR%\;%PATH%;%BASE_DIR%\systemLibs ( в конце %BASE_DIR%)
Как реализовать, чтобы в bat-файле всегда использовался путь - %BASE_DIR%\systemLibs 

2. И главная ошибка, что при запуске у меня собираются dll в каталоге plugins от qt5.15.10, а мне необходимо от qt6.5.1. И соответственно при запуске приложения возникает ошибка о не верности platforms. Пробовал копировать вручную, но это очень муторно, вместо 
c:\msys64\mingw64\share\qt5\plugins\ использовать c:\msys64\mingw64\share\qt6\plugins\
Как сделать правильно, чтобы использовались нужные dll ?

**Comment by EndrII** [comment]
Добрый день, по поводу первого пункта - он уже [исправлен](https://github.com/QuasarApp/CQtDeployer/commit/24afb166380d1c12be532adca5cbe69522052e42). 

по поводу 2го, ошибка выглядит знакомой, мне кажется ч ее уже исправлял, 
Какую версию cqtdeployer вы используете ?

**Reply by AnSMru** [reply]
CQtDeployer version  : 1.6.2311.76d1d20

**Reply by EndrII** [reply]
пришлите лог, с параметром -verbose 3

**Reply by AnSMru** [reply]
[log.zip](https://github.com/QuasarApp/CQtDeployer/files/12451881/log.zip)

**Reply by EndrII** [reply]
У вас были скопированы верные плагины 
```
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/iconengines/qsvgicon.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/imageformats/qgif.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/imageformats/qicns.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/imageformats/qico.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/imageformats/qjp2.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/imageformats/qjpeg.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/imageformats/qmng.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/imageformats/qsvg.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/imageformats/qtga.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/imageformats/qtiff.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/imageformats/qwbmp.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/imageformats/qwebp.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/platforms/qdirect2d.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/platforms/qminimal.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/platforms/qoffscreen.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/platforms/qwindows.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/styles/qwindowsvistastyle.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/tls/qcertonlybackend.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/tls/qopensslbackend.dll
Info: copy :C:/MSYS64/MINGW64/SHARE/QT6/PLUGINS/tls/qschannelbackend.dll
```
я не вижу описанной проблемы в логах.

**Reply by AnSMru** [reply]
Предыдущий лог был, немного с танцами, там была попытка подмены qmake6.exe (6.5.1) вместо qmake.exe (5.15.10)
c:\msys64\mingw64\bin\qmake.exe был заменен файлом c:\msys64\mingw64\bin\qmake6.exe

Высылаю второй лог, когда все в нормальном состоянии (msys2)
[log_qt5.zip](https://github.com/QuasarApp/CQtDeployer/files/12452032/log_qt5.zip)

**Reply by EndrII** [reply]
```
Info: The qmake was found in the PATH variable. qmake : C:/msys64/mingw64/bin/qmake.exe If you want to disable search qmake executable in PATH variable, use the noCheckPATH option

```
У вас в PATH прописан qmake от qt5 
просто укажите путь нужному qmake используя -qmake

**Reply by AnSMru** [reply]
"%CQtDeployer%" -bin prg.exe force-clear deploySystem -qmake C:\msys64\mingw64\bin\qmake6.exe noCheckPATH -verbose 3

Error: Your distribution required Qt libraries but qmake executable cannot be found in the 'qmake' option or RPATH. You are using the option noCheckPATH, please remove this option from your deploy command to search qmake in PATH
Хотя есть три файла по указанному пути C:\msys64\mingw64\bin\ (qmake.exe / qmake6.exe / qmake-qt6.exe), два последних одинаковые.

**Reply by AnSMru** [reply]
Если без опции onCheckPATH, то лог ниже

[log_qt5_01.zip](https://github.com/QuasarApp/CQtDeployer/files/12452247/log_qt5_01.zip)

**Reply by EndrII** [reply]
Такая ошибка может возникать если C:\msys64\mingw64\bin\qmake6.exe  не является файлом или у него нет прав на исполнение, возможно это ссылка.

**Reply by EndrII** [reply]
как вариант еще можно попробовать / сепаратор 
C:/msys64/mingw64/bin/qmake6.exe

**Reply by AnSMru** [reply]
Это windows, ссылка не возможна.
Это первое что пробовал менять \ или /, не работает.

**Comment by EndrII** [comment, marked as answer]
Я нашел в чем проблема. 
Следите за [этим](https://github.com/QuasarApp/CQtDeployer/issues/770) issue, в следующей версии оба пункта из этой дискуссии будут исправлены.

**Reply by AnSMru** [reply]
Спасибо за оперативные консультации.
